### PR TITLE
fix(db) snis dao to set correct array metatable on non-empty arrays

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Fixes `snis` dao to set correct metatable on non-empty arrays:
`cjson.empty_array_mt` to `cjson.array_mt`

### CODE REVIEW

File: kong/db/dao/snis.lua [hunk 1]
```diff
@@ -85,18 +85,26 @@ end
...
     table.insert(name_list, sni.name)
+    name_count = name_count + 1
   end
 
   table.sort(name_list)
 
+  if name_count == 0 then
+    setmetatable(name_list, cjson.empty_array_mt)
+  else
+    setmetatable(name_list, cjson.array_mt)
```
Comment (thibaultcha) [inline kong/db/dao/snis.lua]: You can forget about `cjson.empty_array_mt` and simply stick to setting `array_mt` unconditionally (this could just be a one line change).

**Comment by bungle** [comment]
> Do we have a test case for this fix? It's an important one to have for these tricky edge-cases that have bit us repeatedly.

Test-case added.

File: spec/02-integration/03-db/02-db_core_entities_spec.lua [hunk 2]
```diff
@@ -1951,5 +1954,27 @@ for _, strategy in helpers.each_strategy() do
...
+          local json = cjson.encode(snis)
+          assert.equal("[]", json)
+          assert.equal(cjson.array_mt, getmetatable(snis))
+
+          db.snis:update_list(certificate, {"example.org"})
+          snis = assert(db.snis:list_for_certificate(certificate))
+          json = cjson.encode(snis)
+          assert.equal('["example.org"]', json)
+          assert.equal(cjson.array_mt, getmetatable(snis))
+        end)
```
Comment (thibaultcha) [inline spec/02-integration/03-db/02-db_core_entities_spec.lua]: This test passes with the old `cjson.empty_array_mt` metatable, so long as we remove the `assert.equal(cjson.array_mt, ...)` assertions. I wonder if this is a fix for a known issue (in which we don't JSON-encode the result properly) with an incomplete regression test, or a refactor to use `array_mt` (in which case, it should be bundled with #4724)?

Comment (bungle) [reply spec/02-integration/03-db/02-db_core_entities_spec.lua]: @thibaultcha, the other functions reset the metatable. I will bundle it to #4724, and rebase that to master. I will close this.
